docs/design/server_side_charts.md

```diff
@@ -0,0 +1,44 @@
+
+## Implications
+Note, however, that these assertions require templates to nest, which in turn requires a working dependency management mechanism that supports the definition and resolution of references. 
+
+Currently, DM templates provide such a mechanism. We resolve references iteratively during expansion (i.e., resolve references, expand, repeat). However, a reference resolves to a template file, which is assumed to be the only top level template file in the target folder. It may pull in other files from the folder, but it's the root of the file graph in that folder.
+
+If we preserve this model of a single top level template per folder, then the process of resolving template references doesn't have to change much to use charts instead of raw folders. References can resolve to charts, which we can retrieve to extract their contents (i.e,. since they may be tarballs or zipballs).
+
+If DM can pull files from charts, then there is no need for dependency management at the chart level, as long as we're working with DM templates. This is desirable, at least for the near term, because supporting dependency management at the chart level would require us to:
+* rework the DM dependency management mechanism, and
+* solve a number of problems that the DM mechanism currently solves, such as propagating parameters and expansion results down the expansion graph, for arbitrary templating technologies.
+
+Given the trade-offs here, it seems prudent to deemphasize dependency management at the chart level, and perhaps even to deprecate/remove the mechanism proposed in the chart format document. We've had trouble in the past defining what, exactly, dependencies at the chart level should mean and how we should implement them. In the spirit of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it), we can start by removing them, and then bring them back if and only if, when and only when, we actually need them to satisfy some well defined requirement.
+
+## Compatibility
+It was also observed in discussion on the Slack channel that by focusing only on DM for the moment, we run the risk of locking in assumptions that might cause compatibility problems later. However, if we assume a single top level template per chart, then introducing new templating engines should be strictly additive, and should not require backward incompatible changes to the chart format. The templating engine that owns the top level template in a given chart can have free reign to dictate the chart content constraints and dependency management mechanism for its inputs without affecting the chart formats used by other templating engines, assuming that the overall directory structure and top level `Chart.yaml` remain intact.
```

Comment:
OK I think we're on the same page.  I was previously confused because I was interpreting "at the chart level" to mean "at the granularity of charts" as opposed to meaning "implemented somewhere other than where it is currently implemented in DM".  I agree we should re-use the current implementation, and I'd even go further and say that DM should be the only "template engine" as the term currently means.  In that case, it'd be better called the "chart body expander" or something like that.

I think the existing dependency resolution mechanism is already general enough to cover all the templating technologies we've envisioned.  We can therefore (in future) generalize the backend of expandbird to allow other "templating engines", i.e. other languages, including the go templating already used by Helm.  Then, not only would those extensions not have to worry about implementing dependency mechanisms (this would be done outside of their control), but all charts can use each other no matter what templating technology they use (we definitely want that).